src/Components/Publishing/EditorialFeature/Components/Eoy2018Culture.tsx

```diff
@@ -1,21 +1,306 @@
+import { Box, color, Flex, Sans, Serif, space } from "@artsy/palette"
+import { unica } from "Assets/Fonts"
+import { Author, StyledAuthor } from "Components/Publishing/Byline/Author"
+import { Date, DateContainer } from "Components/Publishing/Byline/Date"
+import { Share } from "Components/Publishing/Byline/Share"
+import { getArticleFullHref } from "Components/Publishing/Constants"
+import { ArticleProps } from "Components/Publishing/Layouts/FeatureLayout"
+import { Nav, NavContainer } from "Components/Publishing/Nav/Nav"
+import { ArticleCards } from "Components/Publishing/RelatedArticles/ArticleCards/ArticleCards"
+import { CaptionContainer } from "Components/Publishing/Sections/Caption"
+import { ImageCollection } from "Components/Publishing/Sections/ImageCollection"
 import {
-  ArticleProps,
-  FeatureLayout,
-} from "Components/Publishing/Layouts/FeatureLayout"
-import { Nav } from "Components/Publishing/Nav/Nav"
+  ImageSetContainer,
+  ImageSetPreview,
+} from "Components/Publishing/Sections/ImageSetPreview"
 import React from "react"
 import styled from "styled-components"
 
-export const Eoy2018Culture: React.SFC<ArticleProps> = props => {
-  return (
-    <ArticleWrapper>
-      <Nav canFix={false} transparent />
-      <FeatureLayout {...props} />
-    </ArticleWrapper>
-  )
+interface State {
+  chapters: any[]
 }
 
+export class Eoy2018Culture extends React.Component<ArticleProps, State> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      chapters: this.makeSectionArrays(),
+    }
+  }
+
+  sectionText = (section, i) => {
+    return (
+      <TextContainer key={i} px={[20, 55]} py={30} mx="auto">
+        <Box width={["100%", "70%"]} mx="auto" maxWidth={1000}>
+          <Serif size="5">
+            <div dangerouslySetInnerHTML={{ __html: section.body }} />
+          </Serif>
+        </Box>
+      </TextContainer>
+    )
+  }
+
+  sectionHeaderText = (section, i) => {
+    return (
+      <SectionHeader key={i}>
+        <div dangerouslySetInnerHTML={{ __html: section.body }} />
+      </SectionHeader>
+    )
+  }
+
+  sectionImageCollection = (section, i) => {
+    return (
+      <ImageWrapper key={i}>
+        <ImageCollection
+          sectionLayout={section.layout}
+          articleLayout="feature"
+          images={section.images}
+          targetHeight={500}
+          gutter={10}
+        />
+      </ImageWrapper>
+    )
+  }
+
+  sectionImageSet = (section, i) => {
+    return (
+      <ImageSetWrapper key={i} px={55}>
+        <ImageSetPreview section={section} />
+      </ImageSetWrapper>
+    )
+  }
+
+  sectionIsHeader = section => {
+    const isText = section.type === "text"
+    const isHeader = isText && section.body.includes("<h1>")
+    return isHeader
+  }
+
+  makeSectionArrays = () => {
+    const {
+      article: { sections },
+    } = this.props
```

Comment:
I'm not sure what you mean, I think this will still be fine to test -- but yeah, haven't gotten around to that yet ... have a ticket to handle tests separately, the implementation is still a bit in flux until we confirm all the content with editorial.

Reply:
I think it's more straightforward to just pass in the `props.article.sections` to `makeSectionArrays` so we don't have to look around to understnad where the data comes from. At a glance it looks mysterious when `makeSectionArrays` is called with no arguments in a `constructor`. What it does is data massagiing, so it would be good to extract it as a plain function.

Also, it has a good amount of complexity that deserves to have its own set of tests, rather than being tested _through_ React.